src/Exceptions/ErrorInfo.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Ably\Exceptions;
+
+use \Exception;
+
+/**
+ * Base class for Ably exceptions, provides read-only access to its fields
+ * @property-read string $message Exception message
+ * @property-read int $code Exception code @see https://github.com/ably/ably-common/blob/master/protocol/errors.json
+ * @property-read int $statusCode HTTP error code
+ */
+class ErrorInfo extends Exception {
```

Comment:
It sounds to me that ErrorInfo is only supposed to be a data structure that we send back when errors occur. It is not in itself an exception (https://github.com/ably/ably-java/blob/870c33991d0a66b182b7f3196d99c21e39d814a1/src/io/ably/types/ErrorInfo.java#L15).

ErrorInfo is supposed to be an attribute of AblyException to investigate what the error details are. https://github.com/ably/ably-java/blob/870c33991d0a66b182b7f3196d99c21e39d814a1/src/io/ably/types/AblyException.java#L107